### PR TITLE
Testacc/packetfabric cs azure dedicated connection

### PR DESCRIPTION
**Pull request by pontovinte**
## Description

ACCTest for packetfabric_cs_azure_dedicated_connection

### CODE REVIEW

File: internal/testutil/test_hcls.go [hunk 1]
```diff
@@ -47,6 +48,13 @@ const CloudRouterBgpSessionType1 = "in"
 const CloudRouterBgpSessionPrefix2 = "192.168.0.0/24"
 const CloudRouterBgpSessionType2 = "out"
 
+// packetfabric_cs_azure_dedicated_connection
+const CloudServiceAzDedDedicatedSubTerm = 1
+const CloudServiceAzDedServiceClass = "metro"
```
Comment (rjouhann) [inline internal/testutil/test_hcls.go:53]: Could you update Service Class to `longhaul`?

**Review by medzin — CHANGES_REQUESTED** [review]

**Comment by medzin** [comment]
I tested it with production API, and it failed with the following error:

```
Error
Failed
Stacktrace
resource_azure_req_express_dedicated_conn_test.go:17: Step 1/1 error: Check failed: Check 1/7 error: Not found: packetfabric_cs_azure_dedicated_connection.pf_702d8546_f4c8_47b5_9453_392b38e3cdd5 in [root]
```

**Comment by rjouhann** [comment]
@pontovinte I just found out you can also disable a dedicated cloud the same way you can disable a port using [/v2/ports/{port_circuit_id}/disable](https://docs.packetfabric.com/api/v2/swagger/#/Ports/disable_port).

Only for `dev`, could you disable the dedicated cloud port before deleting it?

File: internal/provider/resource_azure_req_express_dedicated_conn_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+			testutil.PreCheck(t, []string{
+				testutil.PF_ACCOUNT_ID_KEY,
+			})
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:                    csAzureDedicatedConnectionResult.Hcl,
+				ExpectNonEmptyPlan:        true,
+				Destroy:                   !testutil.IsDevEnv(),
```
Comment (rjouhann) [inline internal/provider/resource_azure_req_express_dedicated_conn_test.go:28]: @pontovinte I just found out you can also disable a dedicated cloud the same way you can disable a port using [/v2/ports/{port_circuit_id}/disable](https://docs.packetfabric.com/api/v2/swagger/#/Ports/disable_port).

Only for `dev`, could you disable the dedicated cloud port before deleting it?

**Comment by pontovinte** [comment]
> > I tested it with production API, and it failed with the following error:
> > ```
> > Error
> > Failed
> > Stacktrace
> > resource_azure_req_express_dedicated_conn_test.go:17: Step 1/1 error: Check failed: Check 1/7 error: Not found: packetfabric_cs_azure_dedicated_connection.pf_702d8546_f4c8_47b5_9453_392b38e3cdd5 in [root]
> > ```
> 
> @pontovinte hard to tell what's wrong here besides the fact it seems the dedicated cloud did not get created. Maybe a problem in the pop selection?

That's hard to tell. But isn't it something related to the Azure account? Some kind of limitation, similar with the one we have with AWS? (just guessing).

**Comment by rjouhann** [comment]
> That's hard to tell. But isn't it something related to the Azure account? Some kind of limitation, similar with the one we have with AWS? (just guessing).

Let me try to run it manually and do some troubleshooting on our side. I will get back to you!

**Comment by rjouhann** [comment]
> > That's hard to tell. But isn't it something related to the Azure account? Some kind of limitation, similar with the one we have with AWS? (just guessing).
> 
> Let me try to run it manually and do some troubleshooting on our side. I will get back to you!

This is the error I am getting when I ran the test locally:
```
resource_azure_req_express_dedicated_conn_test.go:17: Step 1/1 error: Check failed: Check 1/7 error: Not found: packetfabric_cs_azure_dedicated_connection.pf_88cb662d_57f9_4f17_bcbc_8acfc506a39e in [root]
```

When I look in the backend logs, I see the following call made:
1. GET `/v2/locations/cloud` with request_params
`{"cloud_provider": "azure", "cloud_connection_type": "dedicated", "nat_capable": "false", "has_cloud_router": "false", "any_type": "false"}` which look good
2. `GET /v2/locations/CHI4/port-availability`

I am not sure what the problem is but here is how the API call to create a dedicated azure connection look like. Know you do not need special Azure access to create one.

```
{
    "pop": "CHI4",
    "account_uuid": "12345",
    "subscription_term": 1,
    "service_class": "longhaul",
    "zone": "A",
    "speed": "10Gbps",
    "description": "demo-romain",
    "encapsulation": "dot1q",
    "port_category": "primary",
}
```

Bear in mind Azure dedicated cloud do not need the same parameters as port (so no media for example), using the GET `/v2/locations/cloud` with the filters should give you the pop and the zone available to create dedicated connection.

Let me know if this is enough info, otherwise if you could add more debug in `Check 1/7`?

Thanks!

File: internal/provider/resource_azure_req_express_dedicated_conn_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		PreCheck: func() {
+			testutil.PreCheck(t, []string{
+				testutil.PF_ACCOUNT_ID_KEY,
+			})
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:                    csAzureDedicatedConnectionResult.Hcl,
+				ExpectNonEmptyPlan:        true,
```
Comment (rjouhann) [inline internal/provider/resource_azure_req_express_dedicated_conn_test.go:27]: As discussed in other PR, let's make sure we don't skip the terraform plan test. so removing `ExpectNonEmptyPlan`. Thanks!

**Comment by rjouhann** [comment]
Close as duplicate with #425

**Pull request closed without merging**